src/Configuration/RectorConfigBuilder.php

```diff
@@ -699,6 +709,29 @@ public function withTypeCoverageLevel(int $level): self
         return $this;
     }
 
+    /**
+     * @experimental Raise your code quality from the safest rules
+     * to more affecting ones, one level at a time
+     */
+    public function withCodeQualityLevel(int $level): self
+    {
+        $this->isCodeQualityLevelUsed = true;
+
+        $levelRules = LevelRulesResolver::resolve(
+            $level,
+            CodeQualityLevel::RULES,
+            'RectorConfig::withCodeQualityLevel()'
+        );
+
+        $this->rules = array_merge($this->rules, $levelRules);
+
+        foreach (CodeQualityLevel::RULES_WITH_CONFIGURATION as $rectorClass => $configuration) {
```

Comment:
To simplify level management I decided to always run the configured rules (currently there is only one) independently of the level chosen. If you would prefer to do something different, let me know